keras/engine/training.py

```diff
@@ -338,6 +340,46 @@ def weighted(y_true, y_pred, weights, mask=None):
     return weighted
 
 
+def masked_metric(fn):
```

Comment:
Add function that returns a masked metric function given as input a metric function